### PR TITLE
phpstan: fixed calls to Mage_Core_Abstract_Helper

### PR DESCRIPTION
**Pull request by sreichel**
### Description (*)

Still 5.5k lines of supressed [errors](https://github.com/OpenMage/magento-lts/blob/1.9.4.x/.github/phpstan-baseline.neon) ... moving on :metal: 

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All automated tests passed successfully (all builds are green)
 - [x] Add yourself to contributors list

### CODE REVIEW

File: app/code/core/Mage/Adminhtml/Block/Sales/Totals.php [hunk 1]
```diff
@@ -56,8 +58,8 @@ protected function _initTotals()
         /**
          * Add shipping
          */
-        if (!$this->getSource()->getIsVirtual() && ((float) $this->getSource()->getShippingAmount() || $this->getSource()->getShippingDescription()))
-        {
+        if (!$this->getSource()->getIsVirtual()
+            && ((float) $this->getSource()->getShippingAmount() || $this->getSource()->getShippingDescription())) {
```
Comment (kiatng) [inline app/code/core/Mage/Adminhtml/Block/Sales/Totals.php:62]:
```suggestion
        if (!$this->getSource()->getIsVirtual()
            && ((float) $this->getSource()->getShippingAmount() || $this->getSource()->getShippingDescription())
        ) {
```

File: app/code/core/Mage/Adminhtml/Block/Widget/Grid.php [hunk 2]
```diff
@@ -458,16 +483,20 @@ protected function _setFilterValues($data)
...
         if ($this->getCollection()) {
-            $field = ( $column->getFilterIndex() ) ? $column->getFilterIndex() : $column->getIndex();
+            $field = ($column->getFilterIndex()) ?: $column->getIndex();
             if ($column->getFilterConditionCallback() && $column->getFilterConditionCallback()[0] instanceof self) {
                 call_user_func($column->getFilterConditionCallback(), $this->getCollection(), $column);
             } else {
                 $cond = $column->getFilter()->getCondition();
-                if ($field && isset($cond)) {
-                    $this->getCollection()->addFieldToFilter($field , $cond);
+                if ($field) {
```
Comment (kiatng) [inline app/code/core/Mage/Adminhtml/Block/Widget/Grid.php:498]: Is this the same as the original `if ($field && isset($cond)) {`?

File: app/code/core/Mage/Adminhtml/Block/Widget/Grid.php [hunk 3]
```diff
@@ -484,7 +513,7 @@ protected function _addColumnFilterToCollection($column)
      */
     protected function _addLinkModelFilterCallback($collection, $column)
     {
-        $field = ($column->getFilterIndex()) ? $column->getFilterIndex() : $column->getIndex();
+        $field = ($column->getFilterIndex()) ?: $column->getIndex();
```
Comment (kiatng) [inline app/code/core/Mage/Adminhtml/Block/Widget/Grid.php:516]:
```suggestion
        $field = $column->getFilterIndex() ?: $column->getIndex();
```

File: app/code/core/Mage/CatalogSearch/Block/Result.php [hunk 4]
```diff
@@ -53,11 +55,14 @@ protected function _getQuery()
      */
     protected function _prepareLayout()
     {
+        /** @var Mage_CatalogSearch_Helper_Data $hlper */
+        $hlper = $this->helper('catalogsearch');
```
Comment (kiatng) [inline app/code/core/Mage/CatalogSearch/Block/Result.php:59]: Typo? `$helper`.

File: app/code/core/Mage/Customer/Model/Customer.php [hunk 5]
```diff
@@ -884,8 +906,9 @@ public function getTaxClassId()
     /**
      * Check store availability for customer
      *
-     * @param   Mage_Core_Model_Store | int $store
-     * @return  bool
+     * @param Mage_Core_Model_Store | int $store
```
Comment (kiatng) [inline app/code/core/Mage/Customer/Model/Customer.php:909]:
```suggestion
     * @param Mage_Core_Model_Store|int $store
```

File: app/code/core/Mage/Adminhtml/Block/Sales/Totals.php (hunk 1 above)
Comment (sreichel) [reply app/code/core/Mage/Adminhtml/Block/Sales/Totals.php:62]: Style fixes were not planed ... but yes :+1:

File: app/code/core/Mage/Adminhtml/Block/Widget/Grid.php (hunk 2 above)
Comment (sreichel) [reply app/code/core/Mage/Adminhtml/Block/Widget/Grid.php:498]: `$cond` is set one line before as an array ... so `isset($cond)` is useless.

File: app/code/core/Mage/CatalogSearch/Block/Result.php (hunk 4 above)
Comment (sreichel) [reply app/code/core/Mage/CatalogSearch/Block/Result.php:59]: Shame on me  :(

File: app/code/core/Mage/Adminhtml/Block/Widget/Grid.php (hunk 2 above)
Comment (kiatng) [reply app/code/core/Mage/Adminhtml/Block/Widget/Grid.php:498]: But its value can be `null`. No?

Comment (kiatng) [reply app/code/core/Mage/Adminhtml/Block/Widget/Grid.php:498]: From the [manual](https://www.php.net/manual/en/function.isset):

> isset() will return false when checking a variable that has been assigned to null

So, `isset($cond)` is not useless.

File: app/code/core/Mage/Adminhtml/Block/Widget/Grid.php [hunk 6]
```diff
@@ -458,16 +483,20 @@ protected function _setFilterValues($data)
...
         if ($this->getCollection()) {
-            $field = ( $column->getFilterIndex() ) ? $column->getFilterIndex() : $column->getIndex();
+            $field = $column->getFilterIndex() ?: $column->getIndex();
             if ($column->getFilterConditionCallback() && $column->getFilterConditionCallback()[0] instanceof self) {
                 call_user_func($column->getFilterConditionCallback(), $this->getCollection(), $column);
             } else {
                 $cond = $column->getFilter()->getCondition();
-                if ($field && isset($cond)) {
-                    $this->getCollection()->addFieldToFilter($field , $cond);
+                if ($field) {
```
Comment (kiatng) [inline app/code/core/Mage/Adminhtml/Block/Widget/Grid.php:498]: I see. Is this better:

```php
            if ($column->getFilterConditionCallback() && $column->getFilterConditionCallback()[0] instanceof self) {
                call_user_func($column->getFilterConditionCallback(), $this->getCollection(), $column);
            } else {
                $field = $column->getFilterIndex() ?: $column->getIndex();
                if ($field) {
                    $this->getCollection()->addFieldToFilter($field, $column->getFilter()->getCondition());
```

Comment (sreichel) [reply app/code/core/Mage/Adminhtml/Block/Widget/Grid.php]: Think `else` and `if` could combined ... next PR ....

**Review by kiatng — APPROVED** [review]

File: app/code/core/Mage/Checkout/Block/Cart/Sidebar.php [hunk 7]
```diff
@@ -293,16 +298,6 @@ public function getCacheTags()
         );
     }
 
-    /**
-     * Get form key
-     *
-     * @return string
-     */
-    public function getFormKey()
```
Comment (fballiano) [inline app/code/core/Mage/Checkout/Block/Cart/Sidebar.php:301]: can we remove this method?

Comment (sreichel) [reply app/code/core/Mage/Checkout/Block/Cart/Sidebar.php:301]: Y, it is same as in parent `Mage_Core_Block_Abstract::getFormKey()`

Comment (fballiano) [reply app/code/core/Mage/Checkout/Block/Cart/Sidebar.php:301]: true

**Review by fballiano — APPROVED** [review]

**Pull request merged**